### PR TITLE
WFDR-302-UserLanding_Validations

### PR DESCRIPTION
**Pull request by plee99**
## Summary of changes
- small PR where user landing error messages are displayed with better UI
## Trello Task link
https://trello.com/c/OQ66APWW/72-302-userlanding-validations
## Corresponding User Story Number and Title
User case 5: Generate Itinerary 

## Detailed list of changes
modified js to display error messages properly

### CODE REVIEW

**Review by NickMoo5 — CHANGES_REQUESTED** [review]
image
When the error msg shows the location icon in the city textbox doesn't move making it look weird. 

Having the msg appear above the City textbox when it could be meant for the bottom 3 inputs makes it look weird and confusing. Are you able to come up with a solution that indicates to the user exactly what's empty.

**Review by danielqyu — CHANGES_REQUESTED** [review]
the "location" icon moves up when you try submitting with an error.

is it possible to have the error message pop up on the specific missing areas rather than location?
also if possible, the error message should disappear when the user fills in the field(s)

**Comment by plee99** [comment]
- error messaged now indicate to the user exactly what's missing
- icons moving is fixed
- real time errors implemented

image

image

**Review by setaDM — CHANGES_REQUESTED** [review]
Errors look like they work fine and they were updated properly however when I was typing in the city I noticed that your branch doesn't have my autocomplete code.  It appears you removed it. Not sure if it was intentional or if it was a mistake during a merge conflict, but either way you should add it back. If youre unsure about which lines youre missing they are lines 32 and 257 in userLanding.html that you removed.

**Comment by plee99** [comment]
added autocomplete should be good to merge

File: src/main/resources/templates/user/userLanding.html [hunk 1]
```diff
@@ -122,9 +122,6 @@ <h5>We are currently generating your grand adventure, this may take some time</h
 			<div class="d-flex-column text-center mt-2">
 				<h4>Sorry there was an error generating your trip</h4>
 				<h4>Please try again later</h4>
-				<form action="/userLanding" method="get">
-					<button class="btn btn-lg btn-primary mb-2" type="submit">Go back to Trip Planning Page<i class="bi bi-arrow-right ps-3"></i></button>
```
Comment (NickMoo5) [inline src/main/resources/templates/user/userLanding.html:126]: This needs to be added back; seems you deleted from the merge

**Review by NickMoo5 — CHANGES_REQUESTED** [review]
Add back in the code I highlighted in my review as that's needed for some error handling on backend

**Pull request merged**